test/v1beta1/route.go

```diff
@@ -68,7 +68,8 @@ func CreateRoute(t *testing.T, clients *test.Clients, names test.ResourceNames,
 // interval until inState returns `true` indicating it is done, returns an
 // error or timeout. desc will be used to name the metric that is emitted to
 // track how long it took for name to get into the state checked by inState.
-func WaitForRouteState(client *test.ServingBetaClients, name string, inState func(r *v1beta1.Route) (bool, error), desc string) error {
+// ignoreNotFound indicates that we will keep polling even if the route is not found.
+func WaitForRouteState(client *test.ServingBetaClients, name string, inState func(r *v1beta1.Route) (bool, error), desc string, ignoreNotFound bool) error {
```

Comment:
This just doesn't make sense to me. If we have already waited for the Service to be Ready=False there is no way that the Route doesn't exist yet unless something else is going wrong.

Reply:
@dgerd I tried removing this change after I updated the test to use an invalid `RevisionName` in  `TrafficTarget` instead of an invalid image name, and I hit the error `routes.serving.knative.dev "route-not-ready-oddmxzeh" not found` again https://prow.knative.dev/view/gcs/knative-prow/pr-logs/pull/knative_serving/4594/pull-knative-serving-integration-tests/1151260492208017413